packages/dotenv-webpack-plugin/src/index.js

```diff
@@ -0,0 +1,71 @@
+const fs = require("fs");
+const dotenv = require("dotenv");
+const dotenvExpand = require("dotenv-expand");
+const { DefinePlugin } = require("webpack");
+
+class DotenvWebpackPlugin {
+  constructor(config = {}) {
+    const currentDirectory = process.cwd();
+
+    this.config = Object.assign(
+      {},
+      {
+        envFiles: [
+          `${currentDirectory}/.env.example`, // loaded in all cases
+          `${currentDirectory}/.env`, // loaded in all cases
+          `${currentDirectory}/.env.local`, // loaded in all cases, ignored by git
+          `${currentDirectory}/.env.[mode]`, // only loaded in specified mode
+          `${currentDirectory}/.env.[mode].local`, // only loaded in specified mode, ignored by git
+        ],
+        prefixes: ["process.env.", "import.meta.env."],
+      },
+      config,
+    );
+  }
+
+  apply(compiler) {
+    const mode = compiler.options.mode || "production";
+
+    // .local file variables will get precedence
+    let environmentFiles = Array.isArray(this.config.envFiles)
+      ? this.config.envFiles
+      : [this.config.envFiles];
```

Comment:
Need test on this line

Reply:
not possible to reach this case anymore hence removed this line